src/backend/CryptoDoge.BLL/Services/ImagingService.cs

```diff
@@ -1,80 +1,166 @@
+        }
+
+        public async Task DeleteCaffImagesAsync(string caffId)
+        {
+            using var loggerScope = new LoggerScope(logger);
+            var caff = await caffRepository.GetCaffByIdAsync(caffId);
+
+            if (caff != null)
+            {
+                var path = Path.GetFullPath(BasePath);
+                foreach (var ciff in caff.Ciffs)
+                {
+                    var imageName = $"{ciff.Id}.png";
+                    var imagePath = Path.Combine(path, imageName);
+                    File.Delete(imagePath);
+                }
+
+                await caffRepository.DeleteCaffAsync(caff);
+            }
+        }
+
+        public async Task AddCaffCommentAsync(string caffId, string comment)
+        {
+            using var loggerScope = new LoggerScope(logger);
+            var caff = await caffRepository.GetCaffByIdAsync(caffId);
+            if (caff != null)
+            {
+                await caffRepository.AddCaffCommentAsync(new CaffComment
+                {
+                    Comment = comment,
+                });
+            } 
+            else
+            {
+                throw new NotFoundException("Caff does not exists.", 404);
+            }
+        }
+
+        public async Task UpdateCommentOnCaffAsync(string caffCommentId, string comment)
+        {
+            using var loggerScope = new LoggerScope(logger);
+            var caffComment = await caffRepository.GetCaffCommentByIdAsync(caffCommentId);
+            if (caffComment != null)
+            {
+                caffComment.Comment = comment;
+                await caffRepository.UpdateCaffCommentAsync(caffComment);
+            }
+            else
+            {
+                throw new NotFoundException("Caff comment does not exists.", 404);
+            }
+        }
+
+        public async Task DeleteCaffCommentAsync(string caffCommentId)
+        {
+            using var loggerScope = new LoggerScope(logger);
+            var caffComment = await caffRepository.GetCaffCommentByIdAsync(caffCommentId);
+            if(caffComment != null)
+            {
+                await caffRepository.DeleteCaffCommentAsync(caffComment);
+            }
+        }
+
         private Ciff SaveCiff(ParsedCiff parsedCiff)
         {
             using var loggerScope = new LoggerScope(logger);
             var pixels = parsedCiff.Pixels;
             using var bmp = new DirectBitmap(parsedCiff.Width, parsedCiff.Height);
 
-            for (int x = 0; x < pixels.Count; x++)
+            for (int y = 0; y < pixels.Count; y++)
             {
-                for (int y = 0; y < pixels[x].Count; y++)
+                for (int x = 0; x < pixels[y].Count; x++)
                 {
-                    bmp.SetPixel(y, x, Color.FromArgb(pixels[x][y][0], pixels[x][y][1], pixels[x][y][2]));
+                    bmp.SetPixel(x, y, Color.FromArgb(pixels[y][x][0], pixels[y][x][1], pixels[y][x][2]));
```

Comment:
It is worth naming the input parameters according to the function signature of SetPixel. (Y, x, etc ...) instead of (x, y, etc ...) is quite ambiguous.